src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -297,31 +297,19 @@ template <typename TagList, typename Tag>
 using has_subitems = tmpl::not_<
     std::is_same<typename Subitems<TagList, Tag>::type, tmpl::list<>>>;
 
-template <typename TagList, typename Caller, typename Callee, typename List,
-          typename = std::nullptr_t>
+template <typename TagList, typename ComputeTag>
 struct create_dependency_graph {
-  using new_edge =
-      tmpl::edge<DataBox_detail::first_matching_tag<TagList, Callee>,
-                 DataBox_detail::first_matching_tag<TagList, Caller>>;
-  using type = tmpl::conditional_t<tmpl::list_contains_v<List, new_edge>, List,
-                                   tmpl::push_back<List, new_edge>>;
-};
-
-template <typename TagList, typename Caller, typename Callee, typename List>
-struct create_dependency_graph<TagList, Caller, Callee, List,
-                               Requires<is_compute_item_v<Callee>>> {
   using partial_sub_tree =
```

Comment:
I'm not sure this is still the best name for this variable. maybe something like `compute_tag_argument_edges`?